Field.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace gameoflife;
+
+class Field
+{
+    protected $field = [];
+    protected $puffer = [];
+    protected $generationCount = 1;
+    protected $width;
+    protected $height;
+    protected $maxSteps;
+    protected $history = [];
+
+    /**
+     * Field constructor
+     * Constructs an empty field, which can be overwritten by the child constructor.
+     * @param $_width int
+     * @param $_height int
+     * @param $_maxSteps int
+     */
+    public function __construct($_width, $_height, $_maxSteps)
+    {
+        $this->width = $_width;
+        $this->height = $_height;
+        $this->maxSteps = $_maxSteps;
+        for ($y = 0; $y < $this->height; $y++) {
+            for ($x = 0; $x < $this->width; $x++) {
+                $this->field[$x][$y] = 0;
+            }
+        }
+    }
+
+    /**
+     * Print the field
+     * Generates a field and replaces 1->"$" and 0->" ".
+     */
+    public function printField()
+    {
+        for ($y = 0; $y < $this->height; $y++)
+        {
+            for ($x = 0; $x < $this->width; $x++)
+            {
+                echo($this->field[$x][$y] ? "$" : " ");
+            }
+            echo "\n";
+        }
+    }
+
+    /**
+     * Count living neighbours
+     * @param $_x int X-coordinates of the field.
+     * @param $_y int Y-coordinates of the field.
+     * @return int Of the neighbour-count.
+     */
+    function countNeighbours($_x, $_y)
```

Comment:
Is this `private` or `public`? (By default if you omit it, PHP will think it is public)